### ISSUE TITLE
Array slicing beyond the end fails for arrays 32 elements or longer

### ISSUE THREAD

**Issue by jconti** [elm-lang/core#349]
As I noted in the Google Group (https://groups.google.com/d/msg/elm-discuss/8RXblDRVjas/gpEnMLy-BwAJ) I have created a small example of a runtime failure of Array.slice:

```elm
import Array
import List
import Graphics.Element exposing (..)

-- Fails for value of 32, works for 31
aryLast = 32

sliceOffEnd =
  let
    ary = Array.fromList [0..aryLast]
  in
    Array.append (Array.slice 1 aryLast ary) (Array.slice (aryLast + 1) -1 ary)

main : Element
main =
  show sliceOffEnd
```

**Comment by eeue56** [comment]
I've put a fix for this in #399 as I've been hitting the issue a lot with my matrix library.

**Comment by evancz** [comment]
Closing in favor of #649 which will track all array issues and progress on a new implementation.
